cmake/onnxruntime_providers_coreml.cmake

```diff
@@ -7,26 +7,9 @@ endif()
 
 add_compile_definitions(USE_COREML=1)
 
-# Check if we can build the coremltools code for creating an mlpackage with an mlprogram.
```

Comment:
I had a discussion with Scott regarding to this change. The current code is not wrong. 

At runtime ML Program requires: macOS 12+, iOS 15+, watchOS 8+ or tvOS 15+. 

The current code has two problems:
1. It always enables this feature for all macOS target, while our min supported macOS version is 11.0 instead of 12.0.
2. For iOS it should check if the version is >= 15 instead of >=13.